WordPress/gutenberg
# Block Editor: Refactor inner blocks appender components

## What?
PR refactors `ButtonBlockAppender` and `DefaultBlockAppender` components. Changes:

* Remove `withClientId` HoC and use block edit context directly.
* Remove selection for `lastBlockClientId`. Value hasn't been used since #36656.

## Why?
A micro-optimization makes the rendered component tree smaller.

Similar to #60807.

## Testing Instructions
1. Open a post or page.
2. Add a group block.
3. Confirm that the button appender works as before.
4. Replace the button appender with the default appender for the group block. See the patch below.
5. Confirm that the default appender works as before.


## Patch
```diff
diff --git packages/block-library/src/group/edit.js packages/block-library/src/group/edit.js
index a763bc95e60..cf8331caaae 100644
--- packages/block-library/src/group/edit.js
+++ packages/block-library/src/group/edit.js
@@ -118,7 +118,7 @@ function GroupEdit( { attributes, name, setAttributes, clientId } ) {
 	} else if ( ! hasInnerBlocks ) {
 		// When there is no placeholder, but the block is also empty,
 		// use the larger button appender.
-		renderAppender = InnerBlocks.ButtonBlockAppender;
+		renderAppender = InnerBlocks.DefaultBlockAppender;
 	}
 
 	const innerBlocksProps = useInnerBlocksProps(
```

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 
![CleanShot 2024-08-13 at 14 07 47](https://github.com/user-attachments/assets/f0849233-2554-419c-b7e2-4ced5eedae06)

![CleanShot 2024-08-13 at 14 08 37](https://github.com/user-attachments/assets/9352bb1e-45c9-4477-84d6-279f0d628902)
